plugins/src/ios/OptimoveSDKPlugin.swift

```diff
@@ -1,14 +1,51 @@
-@objc(initialize:)
-  func initialize(command: CDVInvokedUrlCommand) {
-    let args = command.arguments
-    
-    let inAppConsentStrategy: InAppConsentStrategy = {
-      if let strategy = args?[2] as? String {
-        return .init(rawValue: strategy)!
-      }
-      return.notEnabled
-    }()
-
-    let config = OptimoveConfigBuilder(optimoveCredentials: args?[0] as? String, optimoveCredentials: args?[1] as? String).enableInAppMessaging(inAppConsentStrategy: inAppConsentStrategy).build()
-    Optimove.initialize(with: config)
-  }
\ No newline at end of file
+import OptimoveSDK
+import NotificationCenter
+@objc(Optimobile_Cordova) class OptimobileSDKPlugin : CDVPlugin {
+    private let internalQueue = DispatchQueue(label: "com.singletioninternal.queue",
+                                              qos: .default,
+                                              attributes: .concurrent)
+    
+    private var pendingPush: PushNotification?
```

Comment:
Can we defer these pieces to the messaging branches?

Reply:
yes